test-runner/adapters/abstract_device_api.py

```diff
@@ -0,0 +1,34 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import six
+import abc
+
+
+@six.add_metaclass(abc.ABCMeta)
+class AbstractDeviceApi:
+
+    @abc.abstractmethod
+    async def connect(self, transport, connection_string, ca_certificate):
+        pass
+
+    @abc.abstractmethod
+    async def disconnect(self):
+        pass
+
+    @abc.abstractmethod
+    async def send_event(self, body):
+        pass
+
+    @abc.abstractmethod
+    async def receive_c2d(self):
+        pass
+
+    @abc.abstractmethod
+    async def enable_methods(self):
```

Comment:
>enable_methods [](start = 14, length = 14)

Is this for enabling device methods in an abstract sense, since it works differently on every SDK?